[Container Images] Split instructions for Container Image and SBOM collection (#22970)
DataDog/documentation: content/en/infrastructure/containers/container_images.md

Datadog collects container image metadata to provide enhanced debugging context for related containers and [Cloud Security Management][8] (CSM) vulnerabilities.

> #### Enable Container Image collection

Reviewer: ```suggestion
#### Enable container image collection
```

{{< tabs >}}
{{% tab "Kubernetes (Operator)" %}}

> Image collection is enabled by default with Datadog Operator version `>= 1.3.0`. If you are using an older version, Datadog recommends you to update it to `1.3.0` or a newer one. </br>

Reviewer: ```suggestion
In Datadog Operator v1.3.0+, image collection is enabled by default. If you are using an older version of the Datadog Operator, Datadog recommends that you update it to v1.3.0+.
```

{{% /tab %}}

{{% tab "Kubernetes (Helm)" %}}

> If you are using Helm version `>= 3.46.0`, image collection is [enabled by default][1].</br>
> Or, add the following to your `values.yaml` Helm configuration file:

Reviewer: ```suggestion
In the Datadog Helm chart v3.46.0+, image collection is [enabled by default][1]. To verify this, or if you are using an earlier Helm chart version, ensure that `datadog.containerImageCollection.enabled` is set to `true` in `datadog-values.yaml`.
```

```yaml
datadog:
  containerImageCollection:
    enabled: true
```
[1]: https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L651
{{% /tab %}}

{{% tab "ECS EC2" %}}

To enable container image collection on your [ECS EC2 instances][1], add the following environment variables to your `datadog-agent` container definition:

```yaml
{
    "containerDefinitions": [
        {
            "name": "datadog-agent",
             ...
            "environment": [
              ...
              {
                "name": "DD_CONTAINER_IMAGE_ENABLED",
                "value": "true"
              }
            ]
        }
    ]
  ...
}
```

[1]: https://docs.datadoghq.com/containers/amazon_ecs/?tab=awscli#setup

{{% /tab %}}

{{% tab "Hosts" %}}

Add the following to your `datadog.yaml` configuration file:

```yaml
container_image:
  enabled: true
```

[1]: /containers/amazon_ecs/?tab=awscli#setup
{{% /tab %}}
{{< /tabs >}}

#### Enable SBOM collection

> The following instructions enable the [Software Bill of Materials][5] (SBOM) collection in the Datadog Agent for CSM Vulnerabilities. SBOM collection enables automatic detection of container image vulnerabilities. Vulnerabilities are evaluated and scanned against your containers every hour. Vulnerability management for container images is included in [CSM Pro and Enterprise plans][10].

Reviewer: ```suggestion
The following instructions turn on [Software Bill of Materials][5] (SBOM) collection for CSM Vulnerabilities. SBOM collection enables automatic detection of container image vulnerabilities. Vulnerabilities are evaluated and scanned against your containers every hour. Vulnerability management for container images is included in [CSM Pro and Enterprise plans][10].
```

**Note**: The CSM Vulnerabilities feature is not available for AWS Fargate or Windows environments.

{{< tabs >}}
{{% tab "Kubernetes (Operator)" %}}

> Add the following to the spec section of your `values.yaml` file:

Reviewer: ```suggestion
Add the following to the spec section of your `datadog-agent.yaml` file:
```

datadog:
  containerImageCollection:
    enabled: true
```
[1]: https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L651
{{% /tab %}}

{{% tab "ECS EC2" %}}

To enable container image collection on your [ECS EC2 instances][1], add the following environment variables to your `datadog-agent` container definition:

```yaml
{
    "containerDefinitions": [
        {
            "name": "datadog-agent",
             ...
            "environment": [
              ...
              {
                "name": "DD_CONTAINER_IMAGE_ENABLED",
                "value": "true"
              }
            ]
        }
    ]
  ...
}
```

[1]: https://docs.datadoghq.com/containers/amazon_ecs/?tab=awscli#setup

{{% /tab %}}

{{% tab "Hosts" %}}

Add the following to your `datadog.yaml` configuration file:

```yaml
container_image:
  enabled: true
```

[1]: /containers/amazon_ecs/?tab=awscli#setup
{{% /tab %}}
{{< /tabs >}}

#### Enable SBOM collection

> The following instructions enable the [Software Bill of Materials][5] (SBOM) collection in the Datadog Agent for CSM Vulnerabilities. This allows you to scan the libraries in container images to detect vulnerabilities. Vulnerabilities are evaluated and scanned against your containers every hour. Vulnerability management for container images is included in [CSM Pro and Enterprise plans][10].

Reviewer 2: It can be misleading with "This allows you.. ", it should probably imply automatic scanning, suggestion below, let's confirm with folks from Documentation and @rajat-luthra-ddog.
```suggestion
The following instructions enable the [Software Bill of Materials][5] (SBOM) collection in the Datadog Agent for CSM Vulnerabilities. SBOM collection enables automatic detection of container image vulnerabilities. Vulnerabilities are evaluated and scanned against your containers every hour. Vulnerability management for container images is included in [CSM Pro and Enterprise plans][10].
```

Author: I agree with your suggestion. I've included it in https://github.com/DataDog/documentation/pull/22970/commits/77eabfd9f2bbd12679b2a708b28d379cfd6a380d.

{{% tab "Kubernetes (Helm)" %}}

> Add the following to your `values.yaml` Helm configuration file:

Reviewer: ```suggestion
Add the following to your `datadog-values.yaml` Helm configuration file:
```
